diff -urd wx2/log.h wx/log.h
--- wx2/log.h	2023-09-29 22:46:06.000000000 +0300
+++ wx/log.h	2024-02-28 14:43:26.944199500 +0200
@@ -1333,7 +1333,7 @@
 // and it will have changed already by then (in fact it even changes when
 // wxString::Format() is called because of vsnprintf() inside it so it can
 // change even much sooner)
-#define wxLOG_KEY_SYS_ERROR_CODE "wx.sys_error"
+#define wxLOG_KEY_SYS_ERROR_CODE wxASCII_STR("wx.sys_error")
 
 #define wxLogSysError \
     wxDO_LOG_IF_ENABLED_WITH_FUNC(Error, MaybeStore(wxLOG_KEY_SYS_ERROR_CODE, \
@@ -1349,7 +1349,7 @@
 #if wxUSE_GUI
     // wxLogStatus() is similar to wxLogSysError() as it allows to optionally
     // specify the frame to which the message should go
-    #define wxLOG_KEY_FRAME "wx.frame"
+    #define wxLOG_KEY_FRAME wxASCII_STR("wx.frame")
 
     #define wxLogStatus \
         wxDO_LOG_IF_ENABLED_WITH_FUNC(Status, MaybeStore(wxLOG_KEY_FRAME).Log)
diff -urd wx2/persist/dataview.h wx/persist/dataview.h
--- wx2/persist/dataview.h	2023-09-29 22:46:06.000000000 +0300
+++ wx/persist/dataview.h	2024-02-28 14:43:26.946068100 +0200
@@ -15,6 +15,7 @@
 #if wxUSE_DATAVIEWCTRL
 
 #include "wx/dataview.h"
+#include "wx/vector.h"
 
 // ----------------------------------------------------------------------------
 // String constants used by wxPersistentDataViewCtrl.
@@ -95,6 +96,7 @@
     virtual bool Restore() wxOVERRIDE
     {
         wxDataViewCtrl* const control = Get();
+        wxVector<wxDataViewColumn*> order(control->GetColumnCount());
 
         for ( unsigned int col = 0; col < control->GetColumnCount(); col++ )
         {
@@ -115,7 +117,32 @@
             if ( RestoreValue(columnPrefix + wxASCII_STR(wxPERSIST_DVC_WIDTH), &width) )
                 column->SetWidth(width);
 
-            // TODO: Set the column's view position.
+            // Retrieve column's view position.
+            int pos;
+            if ( RestoreValue(columnPrefix + wxASCII_STR(wxPERSIST_DVC_POS), &pos) )
+            {
+                if ( pos >= 0 && pos < order.size() && !order[pos] )
+                    order[pos] = column;
+            }
+        }
+
+        // Verify columns' ordering data.
+        for ( auto column: order )
+        {
+            if ( !column )
+            {
+                order.clear(); // do not restore positions
+                break;
+            }
+        }
+
+        // Restore columns' positions.
+        for ( size_t i = 0; i < order.size(); ++i )
+        {
+            auto column = order[i];
+            if ( i != control->GetColumnPosition(column) && control->DeleteColumn(column) ) {
+                control->InsertColumn(i, column);
+            }
         }
 
         // Restore the sort key and order if there is a valid model and sort
